monai/transforms/spatial/dictionary.py

```diff
@@ -131,6 +135,160 @@
 DEFAULT_POST_FIX = PostFix.meta()
 
 
+class SpatialResampled(MapTransform, InvertibleTransform):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.SpatialResample`.
+
+    This transform assumes the ``data`` dictionary has a key for the input
+    data's metadata and contains ``src_affine`` and ``dst_affine`` required by
+    `SpatialResample`. The key is formed by ``key_{meta_key_postfix}``.  The
+    transform will swap ``src_affine`` and ``dst_affine`` affine (with potential data type
+    changes) in the dictionary so that ``src_affine`` always refers to the current
+    status of affine.
+
+    See also:
+        :py:class:`monai.transforms.SpatialResample`
+    """
+
+    backend = SpatialResample.backend
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        mode: GridSampleModeSequence = GridSampleMode.BILINEAR,
+        padding_mode: GridSamplePadModeSequence = GridSamplePadMode.BORDER,
+        align_corners: Union[Sequence[bool], bool] = False,
+        dtype: Optional[Union[Sequence[DtypeLike], DtypeLike]] = np.float64,
+        meta_keys: Optional[KeysCollection] = None,
+        meta_key_postfix: str = DEFAULT_POST_FIX,
+        meta_src_keys: Optional[KeysCollection] = "src_affine",
+        meta_dst_keys: Optional[KeysCollection] = "dst_affine",
```

Comment:
Should we default the keys to `affine` and `original_affine` and describe the common use case in doc-string?
I think for most of the cases, we use this transform to resample image back to the original affine.

Thanks.

Reply:
that is one use case, but I think it is a generic spatial transform and the concepts of 'affine' 'original_affine' are specifically part of the IO in our modules

Reply:
I am OK with both solutions, @ericspod @rijobro what do you guys think?

Thanks.